### PR TITLE
Restructure contracts folder

### PR DESCRIPTION
**Pull request by Sneh1999**
**What changed? Why?**
- Initializing git on create for the user

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: apps/build-onchain-apps/contracts/.github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: test
+
+on: workflow_dispatch
+
+env:
+  FOUNDRY_PROFILE: ci
+
+jobs:
```
Comment (Zizzamia) [inline apps/build-onchain-apps/contracts/.github/workflows/test.yml:8]: Wait, we can have `.github` files in the different part of the app, that's neat!!
Ok I am going to move the other as well ✨

Comment (Zizzamia) [reply apps/build-onchain-apps/contracts/.github/workflows/test.yml:8]: Maybe better question, would it be better to have this file as `apps/build-onchain-apps/.github/~~~`???
So all Workflow related to `apps/build-onchain-apps` are all in the same place?

File: src/utils/git.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { execSync } from 'child_process';
+
```
Comment (Zizzamia) [inline src/utils/git.ts:2]: That's a power move, love this!

File: apps/build-onchain-apps/contracts/.github/workflows/test.yml (hunk 1 above)
Comment (Zizzamia) [reply apps/build-onchain-apps/contracts/.github/workflows/test.yml:8]: Ok, I actually give it a try, and it seems can only be one `.github` and collection of workflows at the root of the project. So I think we need to move this to the way top to have it working.

Comment (Sneh1999) [reply apps/build-onchain-apps/contracts/.github/workflows/test.yml:8]: Thank you! For pointing this out, I will fix this. 

It's still a draft PR though, I still need to get some stuff working, I will let you know once it's done and feel free to review it again then 😄

**Pull request closed without merging**